### PR TITLE
Datepicker: form input props

### PR DESCRIPTION
**Pull request by Zombiefruit**
Task: remove props from the datepicker props that are present in the form input props interface, and implement any missing form input props

Result: added tests for all props in the form input interface and removed duplicate props

### CODE REVIEW

File: test/components/date-picker.spec.tsx [hunk 1]
```diff
@@ -167,6 +167,114 @@ describe('The DatePicker Component', () => {
...
+            const {driver: datePicker, waitForDom, container} = clientRenderer.render(<DatePicker disabled />)
+                .withDriver(DatePickerTestDriver);
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            datePicker.clickOnDatePicker();
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            clientRenderer.render(<DatePicker readOnly />, container);
```
Comment (danielstro) [inline test/components/date-picker.spec.tsx]: When you are rendering here, you don't use with driver. I don't know if this is a problem or not but below you are using it.

File: test/components/date-picker.spec.tsx [hunk 2]
```diff
@@ -167,6 +167,114 @@ describe('The DatePicker Component', () => {
...
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+        });
+
+        it('should not show the dropdown when the date picker is focused and openOnFocus is set to true', async () => {
+            const {driver: datePicker, waitForDom, container} = clientRenderer.render(
+                <DatePicker openOnFocus disabled />
+            ).withDriver(DatePickerTestDriver);
+
+            await waitForDom(() => expect(datePicker.dropDown).to.be.absent());
```
Comment (danielstro) [inline test/components/date-picker.spec.tsx]: Why not use isOpen here?

File: src/components/date-picker/date-picker.tsx [hunk 3]
```diff
@@ -64,6 +64,9 @@ export class DatePicker extends React.PureComponent<DatePickerProps, DatePickerS
                     onChange={this.onInputChange}
                     value={this.state.inputValue}
                     placeholder={this.props.placeholder}
+                    autoFocus={this.props.autoFocus}
+                    name={this.props.name}
+                    required={this.props.required}
```
Comment (JoMarton) [inline src/components/date-picker/date-picker.tsx]: required should be removed until spec exists

File: src/components/date-picker/date-picker.tsx [hunk 4]
```diff
@@ -125,11 +131,12 @@ export class DatePicker extends React.PureComponent<DatePickerProps, DatePickerS
 
     private onInputChange: React.EventHandler<React.SyntheticEvent<HTMLInputElement>> = (event): void => {
         const eventTarget = event.target as HTMLInputElement;
+        this.props.onInput!({value: eventTarget.value});
```
Comment (JoMarton) [inline src/components/date-picker/date-picker.tsx]: Shouldn't there be a test for this new behavior?

File: test/components/date-picker.spec.tsx [hunk 5]
```diff
@@ -167,6 +167,107 @@ describe('The DatePicker Component', () => {
...
+            const {driver: datePicker, waitForDom, container} = clientRenderer.render(<DatePicker disabled />)
+                .withDriver(DatePickerTestDriver);
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            datePicker.clickOnDatePicker();
+
+            await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
+
+            clientRenderer.render(<DatePicker readOnly />, container).withDriver(DatePickerTestDriver);
```
Comment (JoMarton) [inline test/components/date-picker.spec.tsx]: please separate into different tests

**Pull request merged**